### PR TITLE
Upgrade to Foundry v5

### PR DESCRIPTION
**Pull request by connor-baer**
## Purpose

Excerpt from the [release notes](https://github.com/sumup-oss/foundry/releases/tag/v5.0.0):

> Foundry v5 is easier than ever to set up, adapts automatically to your project, and includes a slew of helpful new lint rules.

## Approach and changes

- Upgrade to Foundry v5 and fix all new lint issues
- Upgrade the Next.js template to Foundry v5

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: .browserslistrc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# The browsers that are explicitly supported
+chrome >= 57
+firefox >= 54
+safari >= 10.3
```
Comment (robinmetral) [inline .browserslistrc:4]: Is there a way to state a different version for Safari iOS and mobile? One of them is 10.1 and the other 10.3

File: packages/circuit-ui/cli/index.ts [hunk 2]
```diff
@@ -50,15 +48,14 @@ yargs
...
+    (args) => execute('migrate', args as MigrateArgs),
   )
   .showHelpOnFail(true)
   .demandCommand(1, '')
   .help()
   .version().argv;
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function execute(command: CommandType, args: any): void {
+function execute(command: CommandType, args: MigrateArgs): void {
```
Comment (robinmetral) [inline packages/circuit-ui/cli/index.ts:58]: 🙌

File: packages/cna-template/template/package.json [hunk 3]
```diff
@@ -9,18 +9,18 @@
...
+    "postinstall": "yarn foundry init --presets lint; echo \"Remove the postinstall script from the package.json file after setting up Foundry.\""
   },
   "dependencies": {
     "@emotion/css": "^11.9.0",
     "@emotion/react": "^11.9.0",
     "@emotion/server": "^11.4.0",
     "@emotion/styled": "^11.8.1",
     "@next/bundle-analyzer": "^12.1.6",
-    "@sumup/circuit-ui": "^5.0.0-canary.12",
+    "@sumup/circuit-ui": "^5.0.0",
```
Comment (robinmetral) [inline packages/cna-template/template/package.json:20]: Thanks for updating this as well!

File: packages/cna-template/template/pages/index.tsx [hunk 4]
```diff
@@ -43,11 +43,11 @@ const Page: NextPage = () => (
         </Headline>
         <Body noMargin css={spacing({ bottom: 'giga' })}>
           This is a{' '}
-          <Anchor href="https://nextjs.org" target="_blank" noMargin>
+          <Anchor href="https://nextjs.org" target="_blank">
```
Comment (robinmetral) [inline packages/cna-template/template/pages/index.tsx:46]: Oh yeah. Thank you again

**Review by robinmetral — APPROVED** [review]
Amazing! Looks good to me and that's a fantastic update 🙂 super happy to see the template all shiny again ✨

**Review by amelako — APPROVED** [review]
🚀

File: .browserslistrc (hunk 1 above)
Comment (connor-baer) [reply .browserslistrc:4]: Yes, there's `ios_saf`. Which version is which?

Comment (connor-baer) [reply .browserslistrc:4]: It's Safari 10.1 and iOS Safari 10.3, so this was stricter than necessary. Will update here and in https://github.com/sumup/legacy-browsers.

Comment (connor-baer) [reply .browserslistrc]: Done in 857082d.

**Pull request merged**